crates/bevy_gltf/src/loader.rs

```diff
@@ -392,25 +392,22 @@ async fn load_gltf<'a, 'b, 'c>(
         }
     } else {
         #[cfg(not(target_arch = "wasm32"))]
-        IoTaskPool::get()
-            .scope(|scope| {
-                gltf.textures().for_each(|gltf_texture| {
-                    let parent_path = load_context.path().parent().unwrap();
-                    let linear_textures = &linear_textures;
-                    let buffer_data = &buffer_data;
-                    scope.spawn(async move {
-                        load_image(
-                            gltf_texture,
-                            buffer_data,
-                            linear_textures,
-                            parent_path,
-                            loader.supported_compressed_formats,
-                            settings.load_materials,
-                        )
-                        .await
-                    });
-                });
-            })
+        let futures = gltf.textures().map(|gltf_texture| {
+            let parent_path = load_context.path().parent().unwrap();
+            let linear_textures = &linear_textures;
+            let buffer_data = &buffer_data;
+            load_image(
+                gltf_texture,
+                buffer_data,
+                linear_textures,
+                parent_path,
+                loader.supported_compressed_formats,
+                settings.load_materials,
+            )
+        });
+
+        join_all(futures)
```

Comment:
I don't think join_all runs futures in parallel.

Reply:
That’s surprising. The docs claim it uses FuturesOrdered under the hood for large numbers of futures, and that’s explicitly parallel

Reply:
Oh, I misread it then. The `futures` crate must provide it's own task pool then, which I'm not sure we want. Probably not a big deal in this case, but I'm not sure it's a can of worms we want to open.

Reply:
the actual code for the "small" number of futures is about 30 lines. i can probably pull that in directly if you prefer.

Reply:
i underestimated a bit - a stripped down version comes in at 128 lines including a few unsafe blocks:

[join_all.rs.txt](https://github.com/user-attachments/files/16652054/join_all.rs.txt)

the original doesn't look like much more but `FuturesOrdered` does bring some stream traits and other gubbins. or maybe there's a better alternative way instead?

Reply:
I think I missed the point here. join_all is concurrent but not multi-threaded - it drives all the futures in parallel on the current thread. 

I thought this block was about parallelizing io, but it looks like this is about parallelizing the actual image processing (all data is already loaded), so join_all would be slower.

i'll remove this section (it doesn't matter much to me since it's not io-related anyway).